src/pages/register-a-beacon/activity.tsx

```diff
@@ -26,27 +27,108 @@ interface ActivityOptionsProps extends OptionsProps {
   purpose: string;
 }
 
+const activityMatchingCondition = (activity: Activity) => ({
+  dependsOn: "activity",
+  meetingCondition: (value) => value === activity,
+});
+
+const environmentIsLandMatchingCondition = {
```

Comment:
This is required for only conditionally validating some fields that are displayed if the user has selected the beacon is being used in a `LAND` environment and they have selected the `OTHER` option.